### ISSUE TITLE
Support for Client Credentials Grant

### ISSUE THREAD

**Issue by rgavinc** [cburnette/boxr#114]
## Problem - unsupported feature
When making a call to obtain an access token using the "client_credentials" `grant_type`, the required information is not registered by `Boxr::get_tokens`

```
Boxr::get_tokens(grant_type: 'client_credentials', box_subject_type: 'enterprise', box_subject_id: '123455678')
*** ArgumentError Exception: unknown keywords: box_subject_type, box_subject_id
``` 

## Solution
Allow keywords `box_subject_type` and `box_subject_id` which are required for [Client Credentials](https://developer.box.com/guides/authentication/client-credentials/#how-to-use)

**Comment by xhocquet** [comment]
Hey @rgavinc , thanks for the report. If you're able to write a PR for this that matches our existing patterns, that would be great. Otherwise I'll keep an eye on this when I have some time to dedicate to Boxr!

**Comment by xhocquet** [comment]
Resolved with #120
